plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -2351,4 +2351,42 @@ public function is_created_via( $modus ) {
 	public function untrash(): bool {
 		return (bool) $this->data_store->untrash_order( $this );
 	}
+
+	/**
+	 * Get allowed billing countries for edit.
+	 *
+	 * @return array
+	 */
+	public function get_allowed_billing_countries_for_edit() {
```

Comment:
Please make these two methods a single method that takes the kind of address (`billing` or `shipping`) as a parameter with a sensible default (probably `shipping`) and solid error handling (as in, ensure the argument is one of the two allowed values). There are a few similar methods across the codebase to look at for inspiration.

Reply:
I actually started doing it that way, but then I come up with a complicated and harder to read function.

Anyway, here's a suggestion:
`/**
	 * Get allowed countries for edit.
	 * 
	 * @param  string $address_type Type of address; 'billing' or 'shipping'.
	 * @return array
	 */
	public function get_allowed_countries_for_edit( $address_type = 'billing' ) {
		switch ( $address_type ) {
			case 'billing':
				$countries       = WC()->countries->get_allowed_countries();
				$current_country = $this->get_billing_country();
				break;
			case 'shipping':
				$countries       = WC()->countries->get_shipping_countries();
				$current_country = $this->get_shipping_country();
				break;
			default:
				// Or maybe default to billing, empty array, or all countries ( $countries = WC()->countries->get_countries() ) ?
				return null;
				break;
		}
		if ( ! isset( $countries[ $current_country ] ) ) {
			$all_countries = WC()->countries->get_countries();
			if ( isset( $all_countries[ $current_country ] ) ) {
				$countries[ $current_country ] = $all_countries[ $current_country ];
				if ( apply_filters( 'woocommerce_sort_countries', true ) ) {
					wc_asort_by_locale( $countries );
				}
			}
		}
		return $countries;
	}`

Not sure what to return in the very unlikely change the address type is not billing or shipping:
* default to billing?
* return null?
* return all countries?

Care to comment on the proposed solution, and what to return in when address type is not correct, before me committing it?

Reply:
@lsinger Can I have some feedback on the two issues so that I implement it in an acceptable way?